store/store.go

```diff
@@ -79,17 +79,18 @@ func (s *DefaultStore) SaveBlock(ctx context.Context, block *types.Block, commit
 	}
 	defer bb.Discard(ctx)
 
-	err = bb.Put(ctx, ds.NewKey(getBlockKey(hash)), blockBlob)
-	if err != nil {
-		return err
-	}
-	err = bb.Put(ctx, ds.NewKey(getCommitKey(hash)), commitBlob)
-	if err != nil {
-		return err
-	}
-	err = bb.Put(ctx, ds.NewKey(getIndexKey(block.Height())), hash[:])
-	if err != nil {
-		return err
+	var putErr error // putError contains multiple error if there were issue with multiple 'Put' operatiosn.
+
+	putErr = errors.Join(
+		putErr,
+		bb.Put(ctx, ds.NewKey(getBlockKey(hash)), blockBlob),
+		bb.Put(ctx, ds.NewKey(getCommitKey(hash)), commitBlob),
+		bb.Put(ctx, ds.NewKey(getIndexKey(block.Height())), hash[:]),
+	)
+
+	if putErr != nil {
+		bb.Discard(ctx)
```

Comment:
Note that `defer bb.Discard(ctx)` is called earlier so this is unnecessarly.

Reply:
yeah, missed that